components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2053,19 +2068,92 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             },
         );
 
-        if (centralizedPermissionsEnabled) {
-            log.info("[perms] Checking team operations.", { org: teamId, operations: fineGrainedOps, user: user.id });
-        }
+        const checkAgainstDB = async (): Promise<{ team: Team; members: TeamMemberInfo[] }> => {
+            // We deliberately wrap the entiry check in try-catch, because we're using Promise.all, which rejects if any of the promises reject.
+            const team = await this.teamDB.findTeamById(teamId);
+            if (!team) {
+                // We return Permission Denied because we don't want to leak the existence, or not of the Organization.
+                throw new ResponseError(ErrorCodes.PERMISSION_DENIED, `No access to Organization ID: ${teamId}`);
```

Comment:
Doesn't seem to be a change from this PR, but I think it is more common to return a not-found error code in such cases (including no permissions but resource exists).

Reply:
See https://gitpod.slack.com/archives/C01MLB7J24T/p1675414231569989

Reply:
Yeah, it's not a change from this PR (here, it's just thunked).

Reply:
thanks commented there